main.go

```diff
@@ -304,13 +304,13 @@ func sendReply(w http.ResponseWriter, r *http.Request, tmpl *template.Template,
 	}
 }
 
-// Get the client's real IP address using the X-Real-IP header, or if that is
+// Get the client's real IP address using the X-Forwarded-For header, or if that is
 // empty, http.Request.RemoteAddr. See the sample nginx.conf for using the
-// real_ip module to correctly set the X-Real-IP header.
+// real_ip module to correctly set the X-Forwarded-For header.
 func getClientIP(r *http.Request) (string, error) {
-	xRealIP := r.Header.Get("X-Real-IP")
+	xRealIP := r.Header.Get("X-Forwarded-For")
 	if len(xRealIP) == 0 {
-		log.Warn(`"X-Real-IP" header invalid, using RemoteAddr instead`)
+		log.Warn(`"X-Forwarded-For" header invalid, using RemoteAddr instead`)
```

Comment:
Does testnetfaucet have a sample nginx.conf?

When nginx's realip is setup correctly, it will figure out the actual client IP *from* X-Forwarded-For (when using `real_ip_header X-Forwarded-For;` as you should), and it will *set* `RemoteAddr` so the backend doesn't have to worry about digging through the XFF header.

The only reason you might want to look at `X-Real-IP` in the backend is if you also have `proxy_set_header        X-Real-IP       $remote_addr;` along with the rest of the realip setup, which seems to be common.